docs/index.md

```diff
@@ -0,0 +1,174 @@
+# Getting Started:
+
+```
+$ poetry install
+$ poetry shell
+$ python -m bucket.example
+```
+
+This page describes how to use Bucket to create coverpoints, sample data and merge results. Examples of coverpoints in use can be seen in `example.py`.
+
+## Coverpoints
+
+A coverpoint consists of one or more axes, which are then crossed. Each possible combination of the axes values is called a bucket. Each bucket has a default goal of 10 hits, which can be modified as required, or can be made illegal or ignored. 
+
+Each new coverpoint should inherit from the Coverpoint class, and requires a 'name' and a 'description'. 
+
+``` Python
+class MyCoverpoint(Coverpoint):
+    def __init__(self, name: str, description: str, trigger=None):
+        super().__init__(name, description, trigger=None)
+```
+
+A `setup()` method is then required to create the axes and goals of the coverpoint. This function will only be called if the coverpoint is active. Next, you use `add_axis()` which requires a name, description and values.
+
+| Parameter | Type | Description |
+| --- | ---| ---|
+| name | str | Should be machine readable, containing no spaces. It is suggested to use lower case |
+| description | str | A short description of the coverpoint aim |
+| values | dict,list,tuple,set | All the values/ranges to be covered by the axis |
+
+`Values` can be in the form of numbers or strings. They are processed by the coverpoint when added. A name for each value/range will automatically be generated unless one is given. Names allow more meaning to be given to integers/ranges, while still allowing either the name or the value to be sampled. Names will be shown in the exported coverage.
+
+To specify a named value, a dictionary should be used in the form `{'name': value}`. 
+Alternatively, a list, tuple or set can be passed in and the names will be automatically created.
+Ranges can also be specified by providing a MIN and MAX value in a list in place of a single value.<br>
+eg. `[0,1,2,[3, 9], 10]`
+
+The example below shows an axis being added with five buckets. Four of the buckets only accept a single value, while the remaining one can accept a value from `2` to `13`. Later, when sampling, either the auto-generated name of a bucket can be passed in (eg. "1", "2 -> 13", "15"), or the raw value can be passed instead (eg. 0, 5, 9)
+
+``` Python
+    def setup(self, ctx):
+        self.add_axis(
+            name="my_axis",
+            values=[0, 1, [2, 13], 14, 15],
+            description="Range of values for my_axis",
+        )
+```
+
+----
+
+Goals can be optionally used to name buckets or to define ILLEGAL, IGNORE and modified hit counts. `add_goal` requires a name, number of hits and a description.
+
+| Parameter | Type | Description |
+| --- | --- | ---|
+| name | str | Should be all caps, no spaces. It should also be uniquely named where possible |
+| hits | int | Amount to set goal to. Some special values are accepted (See below)|
```

Comment:
`hits` -> `target` 

`Amount to set goal to` -> `Target number of hits to saturate the bucket`